### ISSUE TITLE
Handle colon style markdown link_attributes

### ISSUE THREAD

**Issue by mcdemarco** [jgm/pandoc#8089]
Some markdown link attribute extensions seem to begin link attributes with a :, but `-f markdown+link_attributes` does not support this style and parses such link attributes as plain text.  For a flavor of markdown that uses the : for this extension, see [Python markdown](https://python-markdown.github.io/extensions/attr_list/).   (While the colon is optional now, I'm working with a codebase that still requires it due to an issue with their tools.)

Tested with pandoc 2.16 and 2.18 on MacOS.

**Comment by mb21** [comment]
Is this a feature request? With what exact input and command would you want what exact output?

**Comment by mcdemarco** [comment]
> Is this a feature request? With what exact input and command would you want what exact output?

Yes, it's a feature request, as, I think, is indicated by the enhancement label.  The input is markdown; the command is `-f markdown+link_attributes` to invoke the `link_attributes` extension.  This extension should ignore a leading `: ` inside the link attribute; e.g.,

    [my link text](my/link/url.html){: #someid .someclass somekey='some value' }

should be treated the same as

    [my link text](my/link/url.html){ #someid .someclass somekey='some value' }

While I am only experiencing this issue with link attributes, Python Markdown permits the colon in all supported attribute lists, and it would not be a bad idea to allow it in related markdown extensions (definitely header_attributes and fenced_code_attributes, and possibly inline_code_attributes and attributes).  I'm not sure what other implementations use the colon this way, so I'm not sure whether attribute lists that don't occur in Python Markdown would need to be supported.

**Comment by jgm** [comment]
I haven't seen this. Is Python Markdown the only implementation that does this?
If so, and if they are now making the colon optional, then I'm hesitant to make changes.

**Comment by mb21** [comment]
Is it the same syntax as Kramdown, we have an issue on that: https://github.com/jgm/pandoc/issues/2711#issuecomment-1014090307

**Comment by jgm** [comment]
I note that the kramdown attribute syntax has many other differences: attributes are placed after blocks, and can occur in many places where pandoc attributes can't.  So, just allowing the colon wouldn't move us any closer to kramdown compatibility.
